certbot-dns-route53/setup.py

```diff
@@ -7,8 +7,8 @@
 version = '1.24.0.dev0'
 
 install_requires = [
-    'boto3',
-    'setuptools>=39.0.1',
+    'boto3>=1.15.15',
```

Comment:
It doesn't need to be constrained here, however, the oldest supported version needed to be updated for our tests to pass. See https://dev.azure.com/certbot/certbot/_build/results?buildId=5130&view=logs&j=61125396-3d92-58eb-3b5e-3d1e52710023&t=e89add51-3cb9-5395-a319-190d6486ad9f&l=653.

This could potentially be relaxed a bit if you're worried about Focal though. As I described [here](https://github.com/certbot/certbot/pull/9216#issuecomment-1050299357), I took the version from RHEL+EPEL 8 and it's possible an older version also works.

Reply:
I only suspect, but don't know for sure, that this would be the only blocker for Focal to package Certbot 1.24.

Based on https://github.com/boto/boto3/commit/8e713130343a784fd4e38de3986b38e87a352f22 and https://packages.ubuntu.com/focal/python3-boto3 it seems like we could lower the constraint to Focal's `1.9.253`.

Reply:
It was good passing thought. Since our distro packages aren't seeing regular updates, we have our own distribution mechanisms, and our team is small, my feeling is we should largely do whatever is easiest for us. With that said, if this dependency is overly strict, we may as well relax it to avoid trouble.

I tested relaxing this to boto3 1.9.253 [here](https://dev.azure.com/certbot/certbot/_build/results?buildId=5214&view=logs&j=61125396-3d92-58eb-3b5e-3d1e52710023&t=e89add51-3cb9-5395-a319-190d6486ad9f&l=610) and the test failed. It looks like boto3 fixed the problem in the commit you linked, however, [the version of botocore that that version of boto3 requires](https://github.com/boto/boto3/blob/1.9.253/setup.py#L17) (since botocore 1.12.253 is the only version that boto3 19.253 supports since [1.12.253 was the last 1.12.* release](https://github.com/boto/botocore/blob/ca9ad21f401a8fb1b55193280f922d0e26cb7e27/CHANGELOG.rst#1130)) did not fix it.

Because of that, I personally just think we should leave things as is here. How about you?

Reply:
Sounds good to me! Thanks for checking.